src/main/java/dev/snowdrop/release/services/BuildConfigUpdateService.java

```diff
@@ -0,0 +1,161 @@
+        try {
+            final String[] releaseVersionMajorMinorFix = releaseVersion.split("\\.");
+            File buildConfigFile = factory.getBuildConfigRelativeTo(repo, releaseVersion);
+            Map<String, String> variableMap = readVariablesFromFile(new FileInputStream(buildConfigFile));
+            variableMap.put("majorVersion", releaseVersionMajorMinorFix[0]);
+            variableMap.put("minorVersion", releaseVersionMajorMinorFix[1]);
+            variableMap.put("patchVersion", releaseVersionMajorMinorFix[2]);
+            variableMap.put("qualifier", qualifier);
+            variableMap.put("milestone", milestone);
+            BuildConfig buildConfigObj = factory.createFromRepo(buildConfigFile);
+            release.getComponents().stream().forEach(component -> {
+//            // TODO: Check if it's a product or a component only template
+//            manageComponentOnly(component);
+                manageProduct(component, variableMap);
+            });
+
+            factory.saveTo(buildConfigObj, generateVariableConfiguration(variableMap), buildConfigFile);
+            return buildConfigFile;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String generateVariableConfiguration(Map<String, String> variableMap) {
+        return variableMap.entrySet()
+            .stream()
+            .map(entry -> "#!" + entry.getKey() + "=" + entry.getValue())
+            .collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private void manageProduct(Component component, Map<String, String> variableMap) {
+        final Issue issue = issueSvc.getIssue(component);
+        final String description = issue.getDescription();
+        final String versionSection = description.substring(description.lastIndexOf("==="));
+        final String[] artifactArr = versionSection.split("\\n[\\n \\r]*");
+        int elementStartingPosition = 0;
+        for (int arrayPos = 0; arrayPos < ((artifactArr.length - 1) / ELEMENTS_IN_GAV_GROUP); arrayPos++) {
+            elementStartingPosition = arrayPos * ELEMENTS_IN_GAV_GROUP + 1;
+            final String gavText = artifactArr[elementStartingPosition];
+            if (GAV_NAME_REGEX_PATTERN.reset(gavText).matches()) {
+                final String gid = GAV_NAME_REGEX_PATTERN.group(2);
+                final String aid = GAV_NAME_REGEX_PATTERN.group(3);
+                final String productVersionText = artifactArr[elementStartingPosition + 2];
+                final String productVersion = productVersionText.split(":")[1].trim();
+                final String supportedVersionText = artifactArr[elementStartingPosition + 3];
+                final String supportedVersion = supportedVersionText.split(":")[1].trim();
+                final var gavProd = toCamelCase(gid) + toCamelCase(aid) + "Prod";
+                if (variableMap.containsKey(gavProd)) {
```

Comment:
What's the point of these checks? This, and the similar code below, should be checked for correctness (it checks nullity of supportedVersion which is already assumed to be not null above) and extracted in a method to be reused in both spots.

Reply:
I have already removed the null checks in the latest version.